test/harry/main/org/apache/cassandra/harry/sut/TokenPlacementModel.java

```diff
@@ -237,6 +237,25 @@ public static List<Node> peerStateToNodes(Object[][] resultset)
         return nodes;
     }
 
+    private static <T> T get(Object[] row, int idx, String name)
+    {
+        T t = (T) row[idx];
+        if (t == null || ((t instanceof Collection<?>) && ((Collection<?>) t).isEmpty()))
+            throw new IncompletePeersStateException(name);
+        return t;
+    }
+
+    /**
+     * When the node sees the new epoch, the update of the peers table is async, so checking the table may yield partial results, so need some way to detect this to enable retries.
```

Comment:
We should make a note of this, and maybe never rely on peers, and instead construct real view of ring every time.

Reply:
well, I think it depends on if we wish to just say "harry only works with the current branch", if so we have a better peers table which is a vtable showing the data in `ClusterMetadata`.

> and instead construct real view of ring every time.

what do you mean here?  would we need to tell harry that we started a token move or would it still query?

Reply:
Right; we would need to tell Harry about the changes in the ring. We already do that in simulator.